### PR TITLE
Fix some static analyzer warnings

### PR DESCRIPTION
**Pull request by tgoyne**

### CODE REVIEW

File: Realm/RLMQueryUtil.mm [hunk 1]
```diff
@@ -731,16 +731,14 @@ Query make_diacritic_insensitive_constraint(NSPredicateOperatorType operatorType
...
                                                              C&& column, T&& value) {
-    if constexpr (is_any_v<C, Columns<Dictionary>>) {
-        // This nesting isnt pretty but without it the compiler will complain about `T` having no known
-        // conversion from Columns<StringData> to Mixed. This is due to the fact that all values on a
-        // dictionary column are boxed in Mixed.
-        if constexpr (is_any_v<T, Mixed, BinaryData, StringData>) {
-            do_add_diacritic_sensitive_string_constraint(operatorType, predicateOptions, std::forward<C>(column), value);
-        }
+
+    if constexpr (is_any_v<C, Columns<Dictionary>> && is_any_v<T, Columns<StringData>, Columns<BinaryData>>) {
```
Comment (dianaafanador3) [inline Realm/RLMQueryUtil.mm:735]: Can we just check if T is mixed and throw otherwise

File: Realm/Tests/MigrationTests.mm [hunk 2]
```diff
@@ -1603,6 +1603,10 @@ - (void)testConvertToEmbeddedAddingMoreLinks {
             XCTAssertNotNil(oldObject);
             XCTAssertNotNil(newObject);
             RLMObject *childObject = newObject[@"object"];
+            XCTAssertNotNil(childObject);
+            if (childObject == nil) {
```
Comment (dianaafanador3) [inline Realm/Tests/MigrationTests.mm:1607]: If the above assertion fails, is this even run?

**Review by dianaafanador3 — APPROVED** [review]
LGTM!! just a few minors comments

File: Realm/Tests/MigrationTests.mm (hunk 2 above)
Comment (tgoyne) [reply Realm/Tests/MigrationTests.mm:1607]: Yes. The rest of the test continues running even if an XCTest assertion fails.

**Review by LowAmmo — APPROVED** [review]

**Comment by LowAmmo** [comment]
Was just about to start making similar changes when I saw this PR!

Thanks for doing this!

**Pull request merged**